lambdas/webhook_processor/webhook_processor.py

```diff
@@ -127,6 +179,224 @@ def _process_results_for_failures(results):
+                jobs_list = EDX_E2E_PR
+
+    return jobs_list
+
+
+def _all_tests_triggered(jenkins_url, sha, jobs_list):
+    """
+    Check to see if the sha has triggered each
+    job in the jobs_list. Looks at both the queue
+    as well as currently running builds
+    """
+    jenkins_username, jenkins_token = _get_credentials_from_s3(jenkins_url)
+
+    queued = _get_queued_builds(
+        jenkins_url, jenkins_username, jenkins_token
+    )
+    running = _get_running_builds(
+        jenkins_url, jenkins_username, jenkins_token
+    )
+    queued_or_running = queued + running
+
+    return _builds_contain_tests(queued_or_running, sha, jobs_list)
+
+
+def _builds_contain_tests(builds, sha, jobs_list):
+    """
+    From the list of all running/queued builds, check
+    to see if the webhook's sha has kicked off every
+    job in the jobs_list
+    """
+    triggered_jobs = []
+    contain_jobs = True
+    if jobs_list:
+        for build in builds:
+            build_job_name = build['job_name']
+            build_sha = build['sha']
+            if (build_job_name in jobs_list
+                    and build_sha == sha
+                    and build_job_name not in triggered_jobs):
+                triggered_jobs.append(build_job_name)
+        if set(triggered_jobs) != set(jobs_list):
+            contain_jobs = False
+
+    return contain_jobs
+
+
+def _parse_executables_for_builds(executable, build_status):
+    """
+    Parse executable to find the sha and job name of
+    queued/running builds.
+    Return list of jobs with the sha that triggered
+    them
+    """
+    builds = []
+    for action in executable['actions']:
+        if 'parameters' in action:
+            for param in action['parameters']:
+                if (param['name'] == 'sha1' or
+                        param['name'] == 'ghprbActualCommit'):
+                    sha = param['value']
+                    if build_status == 'queued':
+                        job_name = executable['task']['name']
+                    elif build_status == 'running':
+                        url = executable['url']
+                        m = re.search(
+                            r'/job/([^/]+)/.*',
+                            urlparse(url).path
+                        )
+                        job_name = m.group(1)
+                    builds.append({
+                        'job_name': job_name,
+                        'sha': sha
+                    })
+    return builds
+
+
+def _get_queued_builds(jenkins_url, jenkins_username, jenkins_token):
+    """
+    Find all builds currently in the queue
+    """
+    builds = []
+    build_status = 'queued'
+
+    # Use Jenkins REST API to get info on the queue
+    # set a timeout for the request to avoid timing out of lambda
+    url = '%s/queue/api/json?depth=0' % (jenkins_url)
+    try:
+        response = get(
+            url,
+            auth=(jenkins_username, jenkins_token),
+            timeout=3
```

Comment:
nit: you can actually define timeout in a tuple of the connect and read timeouts. Let's keep a pattern of setting the connect timeout to 3.05 and the read timeout to whatever was defined by the environment variable.
Reference: http://docs.python-requests.org/en/master/user/advanced/#timeouts